googleanalytics/googleAnalyticsReportTool.py

```diff
@@ -0,0 +1,108 @@
+import os
+from superagi.tools.base_tool import BaseTool
+from pydantic import BaseModel, Field
+import json
+from typing import Type, Optional
+from superagi.resource_manager.file_manager import FileManager
+import yaml
+from google.analytics.data_v1beta import BetaAnalyticsDataClient
+from google.analytics.data_v1beta.types import (
+    DateRange,
+    Dimension,
+    Metric,
+    RunReportRequest,
+)
+
+class GoogleAnalyticsReportToolInput(BaseModel):
+    start_date_of_query: str = Field(..., description="The starting date of the query, in YYYY-MM-DD format")
+    end_date_of_query: str = Field(..., description="The last date of the query, in YYYY-MM-DD format, if today, return today's date")
+    store_report_in_single_file: bool= Field(..., description="True if it is specified that the report should be stored in a single file. False otherwise.")
+
+
+class GoogleAnalyticsReportTool(BaseTool):
+    """
+    Analytics Report Tool
+    """
+    name: str = "Google Analytics Report Tool"
+    args_schema: Type[BaseModel] = GoogleAnalyticsReportToolInput
+    description: str = "Return a google analytics report for the information the user requires"
+    resource_manager: Optional[FileManager] = None
+
+    def _execute(self, start_date_of_query: str, end_date_of_query: str, store_report_in_single_file: bool):
+        property_id = int(self.get_tool_config("PROPERTY_ID"))
+        google_credentials = self.get_tool_config("GOOGLE_CREDENTIALS_FILE")
+
+        self._set_google_credentials(google_credentials)
+
+        client = BetaAnalyticsDataClient()
+        dimensions_and_metrics = self.get_dimensions_and_metrics()
+        filenames = []
```

Comment:
only declare this if store_report_in_single_file is false

Reply:
Checked the code as external agent, also changed a bit of logic to make returning response easier, committing.